gnomad_qc/v4/create_release/create_combined_faf_release_ht.py

```diff
@@ -81,23 +159,55 @@ def _get_pop_meta_indices(
         :return: Indices of populations to keep and their metadata.
         """
         meta = hl.eval(meta)
+        # for adj and raw, but raw is not in faf_meta for genomes release HT,
+        # so we will remove the duplicate idx with list(set(idx))
         idx = [0, 1]
         idx.extend(
             [i for i, m in enumerate(meta) if m.get("pop") in pop_list and len(m) == 2]
         )
-        meta = [meta[i] for i in idx]
+        meta = [meta[i] for i in list(set(idx))]
 
         return idx, meta
 
     logger.info("Keeping only frequencies that are needed (adj, raw, adj by pop)...")
+    # if faf and faf_meta doesn't exist, then we calculate it with freq
+    # TODO: remove this once v4 exomes freq_ht is finalized
+    if "faf_meta" not in ht.globals.keys():
+        faf, faf_meta = faf_expr(
+            ht.freq, ht.freq_meta, ht.locus, pops_to_exclude=faf_pops_to_exclude
+        )
+        faf_meta_by_pop = {
+            m.get("pop"): i for i, m in enumerate(faf_meta) if m.get("pop")
+        }
+        faf_meta_by_pop = hl.literal(faf_meta_by_pop)
+
+        # Compute group max (popmax)
+        grpmax = pop_max_expr(
+            ht.freq, ht.freq_meta, pops_to_exclude=faf_pops_to_exclude
+        )
+        grpmax = grpmax.annotate(faf95=faf[faf_meta_by_pop.get(grpmax.pop)].faf95)
+
+        ht = ht.annotate(faf=faf, grpmax=grpmax)
+        ht = ht.annotate_globals(faf_meta=faf_meta)
+
+    # Rename popmax to grpmax because it's called 'popmax' in v3 release HT.
+    # TODO: remove this once v4 genomes freq_ht is finalized
+    if "popmax" in ht.row.keys():
+        ht = ht.transmute(grpmax=ht.popmax)
+
     freq_idx, freq_meta = _get_pop_meta_indices(ht.freq_meta, pops)
     faf_idx, faf_meta = _get_pop_meta_indices(ht.faf_meta, faf_pops)
 
+    # Compute FAF max (fafmax)
+    fafmax = pop_max_for_faf_expr(ht.faf, ht.faf_meta)
+    ht = ht.annotate(fafmax=fafmax)
```

Comment:
Just confirming this was discussed and needed in the combined FAF? If so maybe just shorten it to a single line, no need to assign fafmax before the annotation.